text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/ShowEntitySerializer.java

```diff
@@ -60,9 +65,10 @@ public HoverEvent.ShowEntity read(final JsonReader in) throws IOException {
 
     while (in.hasNext()) {
       final String fieldName = in.nextName();
-      if (fieldName.equals(SHOW_ENTITY_TYPE)) {
+      // TODO make this more flexible
```

Comment:
I cant think of a way to make this support old and new rn, need input

Reply:
In any case, I don't think this can be done in backwards-compatible way.

We should probably create a wrapper for `HoverEvent.ShowItem` and `HoverEvent.ShowEntity`, and serialize that instead. Wrapper would include all necessary context for Serializers to work with.